src/api/admin/types.ts

```diff
@@ -429,6 +427,12 @@ export type Location =
        * Get file from URL
        */
       url: string;
+    }
+  | {
```

Comment:
This in particular feels weird to me -- an empty struct as a union variant. It's only there because `Location` gets intersected with `AppRoleDnaManifest`, because the corresponding stuff (that is, the optional `AppRoleDnaManifest::location` field's enum variant value) gets collapsed into `AppRoleDnaManifest` and we've got to find some similar way of modelling that.